model/scheduler/SchedulerConfigStorage.php

```diff
@@ -0,0 +1,168 @@
+
+use oat\oatbox\cache\SimpleCache;
+use oat\taoNccer\model\registry\eventAnonymizer\AbstractEventAnonymizer;
+use oat\taoScheduler\model\job\JobInterface;
+use oat\taoScheduler\model\job\Job;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+use oat\taoScheduler\model\SchedulerException;
+use Psr\SimpleCache\InvalidArgumentException;
+use common_ext_ExtensionsManager as ExtensionsManager;
+use common_ext_Extension as Extension;
+
+/**
+ * Class SchedulerConfigStorage
+ * Storage fetches scheduled jobs from installed extensions configuration
+ *
+ * @package oat\taoScheduler\model\scheduler
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+class SchedulerConfigStorage implements SchedulerStorageInterface
+{
+    use ServiceLocatorAwareTrait;
+
+    const CACHE_KEY = 'SchedulerJobsStorage';
+    const MANIFEST_KEY = 'scheduledJobs';
+
+    /**
+     * @inheritDoc
+     * @param JobInterface $job
+     * @return bool|void
+     * @throws InvalidArgumentException
+     * @throws SchedulerException
+     */
+    public function add(JobInterface $job)
+    {
+        $jobToAdd = json_encode($job);
+        if ($this->isExists($job)) {
+            throw new SchedulerException('Job already exists');
+        }
+
+        $jobs = $this->getEncodedJobs();
+        $jobs[] = $jobToAdd;
+        return $this->getPersistence()->set(self::CACHE_KEY, json_encode($jobs));
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function remove(JobInterface $job)
+    {
+        $jobToDelete = json_encode($job);
+        if (!$this->isExists($job)) {
+            throw new SchedulerException('Job does not exist');
+        }
+
+        $jobs = $this->getEncodedJobs();
+        $key = array_search($jobToDelete, $jobs);
+        unset($jobs[$key]);
+        return $this->getPersistence()->set(self::CACHE_KEY, json_encode($jobs));
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getJobs()
+    {
+        $jobs = [];
+        foreach ($this->getEncodedJobs() as $job) {
+            $jobs[] = Job::restore($job);
+        }
+        return $jobs;
+    }
+
+    /**
+     * Check if job exists in the storage
+     * @param JobInterface $job
+     * @return bool
+     */
+    private function isExists(JobInterface $job)
+    {
+        $jobs = $this->getEncodedJobs();
+        return in_array(json_encode($job), $jobs);
+    }
+
+    /**
+     * @return SimpleCache
+     */
+    private function getPersistence()
+    {
+        return $this->getServiceLocator()->get(SimpleCache::SERVICE_ID);
+    }
+
+    /**
+     * Initialize log storage
+     *
+     * @param $persistence
+     * @return \common_report_Report
+     */
+    public function install()
+    {
+        return new \common_report_Report(\common_report_Report::TYPE_SUCCESS, __('Cache scheduler storage installed'));
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function refreshJobs(): bool
+    {
+        $this->getPersistence()->delete(self::CACHE_KEY);
```

Comment:
Which inconsistency do you mean? 
Please not that this operation will not be executed in parallel or different instances.

Reply:
We run update only on one instance and then synchronize configs.
Tao updater is not supposed to be run in parallel. It may lead to unpredictable issues.